File: ios/Classes/bridge/AdropChannel.swift
import Foundation

struct AdropChannel {
    static let invokeChannel = "io.adrop.adrop-ads"
    static let bannerEventListenerChannel = "\(invokeChannel)/banner"
    static let nativeEventListenerChannel = "\(invokeChannel)/native"
    
    static func bannerEventListenerChannelOf(id: String) -> String {
        return "\(bannerEventListenerChannel)_\(id)"
    }
    
    static func adropEventListenerChannel(adType: AdType, id: String) -> String? {
        switch adType {
        case AdType.interstitial:
            return "\(invokeChannel)/interstitial_\(id)"
        case AdType.rewarded:
            return "\(invokeChannel)/rewarded_\(id)"
        case AdType.popup:
            return "\(invokeChannel)/popup_\(id)"
        case AdType.native:
            return "\(invokeChannel)/native_\(id)"
        case AdType.undefined:
            return nil
        }
    }
}

File: lib/src/bridge/adrop_channel.dart
import 'package:adrop_ads_flutter/src/adrop_ad.dart';

class AdropChannel {
  static const invokeChannel = "io.adrop.adrop-ads";
  static const bannerEventListenerChannel = "$invokeChannel/banner";
  static const nativeEventListenerChannel = "$invokeChannel/native";

  static String bannerEventListenerChannelOf(int id) =>
      "${bannerEventListenerChannel}_$id";
  static String? adropEventListenerChannelOf(AdType adType, String id) {
    switch (adType) {
      case AdType.interstitial:
        return "$invokeChannel/interstitial_$id";
      case AdType.rewarded:
        return "$invokeChannel/rewarded_$id";
      case AdType.popup:
        return "$invokeChannel/popup_$id";
      case AdType.native:
        return "$invokeChannel/native_$id";
      case AdType.undefined:
        return null;
    }
  }
}
